google-cloud-firestore/src/main/java/com/google/cloud/firestore/StreamableQuery.java

```diff
@@ -331,7 +379,7 @@ public void onError(Throwable throwable) {
 
               startAfter(cursor)
                   .internalStream(
-                      runQueryResponseObserver,
+                      streamResponseObserver,
```

Comment:
we're re-using the `streamResponseObserver`, which is great. This is good since it'll end up with only 1 first-response-latency. We should verify (using a test) the following two scenarios:

```
doc1, doc2
retryable_error_1
retry with cursor
doc3
retryable_error_2
retry with cursor
doc4
completed
```

And we should verify that the following occurs:
```
doc1
streamResponseObserver.onNext --> records first-response latency
doc2
streamResponseObserver.onNext
retryable_error_1
doc3
streamResponseObserver.onNext
retryable_error_2
doc4
streamResponseObserver.onNext
streamResponseObserver.onCompleted
// assert that streamResponseObserver.onCompleted gets called exactly once
// assert that streamResponseObserver.onError does not get called
```

And the second scenario would be:

```
doc1, doc2
non-retryable-error
```
And we should verify that the following occurs:
```
doc1
streamResponseObserver.onNext --> records first-response latency
doc2
streamResponseObserver.onNext
non-retryable-error
streamResponseObserver.onError
// assert that streamResponseObserver.onCompleted does not get called
// assert that streamResponseObserver.onError gets called exactly once.
```

Reply:
it should have only one first-response-latency, as the observer object itself has a `receivedFirstResponse` flag planted inside. Since we are passing down the same object, the flag should get flipped only once. 

i haven't looked into testing yet, but seems like we can mock above responses with `doAnswer`.

Reply:
Ok, I've filed b/376308036. We should add these tests as part of your testing PR.